src/CompositeShape.cpp

```diff
@@ -76,8 +76,8 @@ void CompositeShape::addShape(const Shape& shape, const Pose3D& pose) {
     bb_ = Box(min_, max_);
 }
 
-const Box CompositeShape::getBoundingBox() const {
-    return bb_;
+Box CompositeShape::getBoundingBox() const {
+    return Box(bb_);
```

Comment:
No need to make a copy (ie you're copying it twice now before it is in the user-supplied destination.